src/mpsse-cli.cpp

```diff
@@ -0,0 +1,200 @@
+    APP_CHECK_STATUS(ftStatus);
+    printf("Number of available MPSSE channels: %d\n", (int) channels);
+
+    return channels;
+}
+
+static void printMPSSEchannelInfo(int channels) {
+    FT_STATUS ftStatus;
+    FT_DEVICE_LIST_INFO_NODE devList = {0};
+    // Iterate through the channels and print their info
+    if (channels > 0) {
+        for (uint32 i = 0; i < channels; i++) {
+            ftStatus = SPI_GetChannelInfo(i, &devList);
+            APP_CHECK_STATUS(ftStatus);
+            printf("Information on channel number %d:\n", i);
+            /* print the dev info */
+            printf("    Flags=0x%x\n", devList.Flags);
+            printf("    Type=0x%x\n", devList.Type);
+            printf("    ID=0x%x\n", devList.ID);
+            printf("    LocId=0x%x\n", devList.LocId);
+            printf("    SerialNumber=%s\n", devList.SerialNumber);
+            printf("    Description=%s\n", devList.Description);
+            // broken in cpp 20, but since we will be moving away from the properitary lib I will not be fixing it.
+//            printf("    ftHandle=0x%x\n", (unsigned int) devList.ftHandle); /*is 0 unless open*/
+        }
+    }
+}
+
+int CB_printCliVersion(arg_t *arg) {
+    printf("0.1.0\n");
+    return 0;
+}
+
+int CB_printFTDIdevices(arg_t *arg) {
+    printMPSSEchannelInfo(getMPSSEchannelCount());
+    return 0;
+}
+
+int main(int argc, char *argv[]) {
+    int retVal = 0;
+    bool useProtocol = false;
+    Protocol *IO;
+    arg_t arg = {
+            .clk = 100000,
+            .channel = 0,
+            .len = 1,
+    };
+
+    // prase the cli to a usable object
+    if (parsecli(argc, argv, &arg) < 0) {
+        printf("Please provide at least one option and value.\n\n");
+        // print help screen to help the user
+        arg.type = ARG_HELP;
+    }
+
+    // call correct function for subcommand or option
+    switch (arg.type) {
+        case ARG_SPI:
+            useProtocol = true;
+            // Check if the FTDI serial module is loaded. If so, remove it.
+            // This requires sudo when running after a build. Not required when installed.
+            if (checkIfFtdiModuleLoaded() > 0) {
+                removeFtdiModule();
+            }
+
+            /* init library */
+            Init_libMPSSE();
+            IO = new SPI();
+            break;
+        case ARG_I2C:
+            useProtocol = true;
+            IO = new I2C();
+            break;
+        case ARG_HELP:
+            retVal = CB_printCLIoptions(&arg);
+            break;
+        case ARG_VERSION:
+            retVal = CB_printCliVersion(&arg);
+            break;
+        case ARG_LIST:
+            retVal = CB_printFTDIdevices(&arg);
+            break;
+        case ARG_NONE:
+        default:
+            // We made it this far.. Assume no valid argument was given.
+            printf("Invalid arguments given.\n");
+            retVal = -1;
+    }
+    if (useProtocol) {
+
```

Comment:
I suggest we check for available channels here to prevent any issues later in the app. I added a comment in the I2c write function that seg faults if no FTDI devices are connected.